app/services/order_checkout_restart.rb

```diff
@@ -13,7 +13,7 @@ def call
     clear_shipments
     clear_payments
 
-    order.reload
+    order.reload.update!
```

Comment:
why is this needed?

Reply:
This is related to callbacks hell with adjustments. Previously, the payment fee adjustment was being updated in this step, and a callback on adjustments (which only affects adjustments directly on the order) was calling `order.update!`.

Moving the payment fee adjustment to the payment means that callback is now not triggered, which means `order.update!` is not called automatically. In this case we're changing the order's contents (in the lines above) and `order.update!` actually does need to be called, or the totals will not be updated.